client/internal/inmempackagestore.go

```diff
@@ -92,18 +96,22 @@ func (l *InMemPackagesStore) SetAllPackagesHash(hash []byte) error {
 }
 
 func (l *InMemPackagesStore) GetContent() map[string][]byte {
-	return l.fileContents
+	return maps.Clone(l.fileContents)
 }
 
 func (l *InMemPackagesStore) GetSignature() map[string][]byte {
 	return l.fileSignatures
 }
 
 func (l *InMemPackagesStore) LastReportedStatuses() (*protobufs.PackageStatuses, error) {
+	l.statusLock.RLock()
+	defer l.statusLock.RUnlock()
 	return l.lastReportedStatuses, nil
 }
 
 func (l *InMemPackagesStore) SetLastReportedStatuses(statuses *protobufs.PackageStatuses) error {
+	l.statusLock.Lock()
 	l.lastReportedStatuses = statuses
+	l.statusLock.Unlock()
```

Comment:
Nit: the function above uses the `defer` form. Would be nice to do the same here for consistency.